test/integration/test/philips_color_test.js

```diff
@@ -0,0 +1,144 @@
+    });
+  });
+
+  describe('manipulate lights', function () {
+    before(function() {
+      // turn off all lights in the simulator
+      Prepper.philipshue_server.turnOffLight(1);
+      Prepper.philipshue_server.turnOffLight(2);
+      Prepper.philipshue_server.turnOffLight(3);
+
+      return chakram.put(Prepper.foxboxManager.getterURL,getterPayload)
+      .then(function(listResponse) {
+        lights = Object.keys(listResponse.body);
+        expect(lights.length).equals(3);
+        expect(listResponse).to.have.status(200);
+      });
+    });
+
+    // Currently, there is no mapping between the foxbox
+    // id and the philips hue id until the tag feature is implemented
+    it('Turn on all lights', function(){
+      var lightID = lights[0].replace('getter','setter');
+      var payload = {'select': {'kind': 'LightOn'}, 
+      'value': { 'OnOff': 'On' } };
+      
+      return chakram.put(Prepper.foxboxManager.setterURL,payload)
+      .then(function(cmdResponse) {
+        
+        expect(cmdResponse).to.have.status(200);
+        expect(cmdResponse.body[lightID]).equals(null);
+        return chakram.put(Prepper.foxboxManager.getterURL,getterPayload);
+     })
+      .then(function(listResponse){
+        expect(listResponse).to.have.status(200);
+        lights = Object.keys(listResponse.body);
+        expect(lights.length).equals(3);
+
+        lights.forEach(light => {
+          expect(listResponse.body[light].OnOff).equals('On'); 
+        });
+      });
+    });
+    
+    var testParams = [{testName:'upper boundary',
+    lightID:0, hue: 359, sat:0.9, bri:0.9}, 
+    {testName:'middle', lightID:1, hue: 200, sat:0.5, bri:0.5},
+    {testName:'lower boundary', lightID:2, hue: 0, sat:0.1, bri:0.1},
+    ];
+
+    testParams.forEach(testParam => {
+      it(testParam.testName, function() {
+      
+        lights[testParam.lightID] = 
+        lights[testParam.lightID].replace('getter:power','setter:color');
+        var hue = testParam.hue;
+        var sat = testParam.sat;
+        var bri = testParam.bri;
+        
+        // send various values within valid range
+        var payload = {'select': {'id': lights[testParam.lightID]}, 
+        'value': { 'Color': {'h':hue,'s':sat,'v':bri} } };
+
+        return chakram.put(Prepper.foxboxManager.setterURL,payload)
+        .then(function(cmdResponse) {
+          expect(cmdResponse).to.have.status(200);
+          var HSB = Prepper.philipshue_server.
+          getHSB(Prepper.philipshue_server.lastLight());
+          var trans = translateHSB(hue,sat,bri);
+          expect(HSB.h).equals(trans.h);
+          expect(HSB.s).equals(trans.s);
+          expect(HSB.b).equals(trans.b);             
+        });
+      });
+    });
+
+    it('Send invalid sat value', function() {   
+      lights[0] = lights[0].replace('getter:power','setter:color');
+      var hue = 200;
+      var sat = 1.5;  // Value should be less than or equal to 1 
```

Comment:
Nit: I'd explicitly name the variable that holds the incorrect value. The others should be shared with the other tests, I believe